### DISCUSSION TITLE
Database Backup & Restore

### DISCUSSION THREAD

**Discussion by Jurgens92** [kimai/kimai#5118] [Questions & Answers] [answered]
Hi,

is it possible to do a backup from kimai 1.11 on Ubuntu 18 (php7.4, composer 1)

and restore that database to a fresh install of Kimai (latest 2.23)

**Comment by kevinpapst** [comment]
Likely, but you have to test it.

**Reply by Jurgens92** [reply]
Thanks. 

will test and revert back

**Reply by Jurgens92** [reply]
I have done the restore, however I just get a 500 internet error

I took it a bit further and did some troubleshooting, which lead me to try and restore a single table

same thing.

the particular table that I restored was: kimai2_customers 

also, what I tried was, to restore the original (new install) database back, with no avail, and got the same 500 error

so that made me think, im doing something wrong.
also note, I did the bin/console kimai:reload --env=prod command as well as the permissions.

**Reply by Jurgens92** [reply]
[prod.log](https://github.com/user-attachments/files/17398905/prod.log)

**Reply by kevinpapst** [reply]
These are all file permissions errors, see docs on that topic.

**Comment by Jurgens92** [comment]
So, I've struggled a lot with this.

I ended up using HeidiSQL to export and import the tables,

seems to work thus far

**Reply by kevinpapst** [reply]
The only database backup that works on 100% of all cases is the command line tool "mysqldump".

**Comment by Jurgens92** [comment, marked as answer]
Ahhh I see
thanks.

I managed to get it working

-----------
bin/console doctrine:migrations:status
bin/console doctrine:migrations:migrate
bin/console doctrine:schema:validate
bin/console doctrine:migrations:sync-metadata-storage
bin/console doctrine:migrations:status
bin/console doctrine:schema:update --force
bin/console doctrine:schema:validate
------------


however, when I try to create a new entry, once I select a customer, then an internal server error appear.
it allows me to select the customer, but "project" is greyed out,
however, when I navigate to any customer, there are projects listed within the edit menu.
even when I create a dummy "test" project for that client, the problem persist

**Reply by Jurgens92** [reply]
I just re-did everything, and then it worked.

thanks

all good now

**Reply by kevinpapst** [reply]
If you ran `bin/console doctrine:schema:update --force` you are out of sync with Kimai and the next updates will at some point most certainly fail. When you then ask for help, it will be impossible to support you.
The docs never mention this command for a very strong reason.

The command `bin/console doctrine:migrations:sync-metadata-storage` is also rather dangerous and should be avoided at all costs.

**Reply by Jurgens92** [reply]
okay,
well no worries.

what I ended up doing is, I took the migrated version's DB,
create a backup and restored it to a fresh install of Kimai 2.23, on ubuntu 24.04

its a workaround but it worked.

I should not have any future problems now, will I?
